### PR TITLE
Connected Login, Register, and Logout actions with API

### PR DESCRIPTION
**Pull request by thomasweng15**
Also ditched initialState and used the Immutable.js store that you set up. 

Users can now register, login, and logout. Will need to add success and error flash messages, but the actions themselves work now.

### CODE REVIEW

**Comment by thomasweng15** [comment]
I am using empty strings instead of null because null can lead to issues on input forms. There may be a more elegant solution to this but I'm leaving them as empty strings for now.

File: src/main/webapp/front-end/src/components/Login.jsx [hunk 1]
```diff
@@ -27,4 +51,28 @@ export default class Login extends React.Component {
         </form>
       </Grid>
   }
-}
\ No newline at end of file
+}
+
+const mapStateToProps = state => {
+  return {
+      email: state.get('loginEmail').get('email'),
```
Comment (smatt989) [inline src/main/webapp/front-end/src/components/Login.jsx]: .get is used for only going 1 level deep into an immutable js object.
.getIn is used for going multiple levels deep:
`state.getIn(['loginEmail', 'email'])`
also, you can provide a default value by adding a second argument:
`state.getIn(['loginEmail', 'email'], null)`

File: src/main/webapp/front-end/src/components/Register.jsx [hunk 2]
```diff
@@ -2,22 +2,45 @@ import React from 'react';
...
+    super(props);
+    this.state = {
+      redirectToReferrer: false
+    }
+
+    this.onSubmit = (e) => {
+      e.preventDefault();
+      this.props.onSubmit(this.props.email, this.props.password)
+        .then(response => {
+          this.setState({ redirectToReferrer: response })
```
Comment (smatt989) [inline src/main/webapp/front-end/src/components/Register.jsx]: ok interesting.  we could conceivably say that all states will be immutable js objects.  or we could have local states be simple js objects.  don't really care.  just pointing out the discrepancy.

Comment (thomasweng15) [reply src/main/webapp/front-end/src/components/Register.jsx]: Sure, I was just sticking close to react router example code here -- I think in certain cases like this one local state makes sense, if it doesn't affect the rest of the codebase

File: src/main/webapp/front-end/src/reducer.js [hunk 3]
```diff
@@ -28,7 +32,11 @@ function cleanState() {
         markingImageSeen: Map({error: null, loading: false}),
         currentImage: Map({image: null, loadConfigs: null, markedSeen: false, error: null, loading: false}),
         savingLabels: Map({error: null, loading: false}),
-        currentLabels: Map({labels: List.of(), error: null, loading: false})
+        currentLabels: Map({labels: List.of(), error: null, loading: false}),
+        signupEmail: Map({ email: '' }),
+        signupPassword: Map({ password: '' }),
+        loginEmail: Map({ email: '' }),
+        loginPassword: Map({ password: '' })
```
Comment (smatt989) [inline src/main/webapp/front-end/src/reducer.js:39]: what is the best thinking on setting up state?  is it good to have flat properties like this?  more nested?  just interesting..

File: src/main/webapp/front-end/src/reducer.js [hunk 4]
```diff
@@ -341,14 +349,40 @@ function viewParticipantImageLabelsError(state, error) {
     return state.set('currentLabels', Map({labels: List.of(), error: Immutable.fromJS(error), loading: false}))
 }
 
+function signupEmailChanged(state, email) {
+    return state.set('signupEmail', Map({ email: Immutable.fromJS(email) }))
```
Comment (smatt989) [inline src/main/webapp/front-end/src/reducer.js]: this is a little silly.  email is a string.  Immutable.fromJS(email) == email.

File: src/main/webapp/front-end/src/utilities.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
+import store from './store.js';
+import { login, loginError, loginSuccess, loginClearInputs } from './actions.js';
 var _ = require('lodash');
 
 export var authenticatedSession = null;
 
-export var authenticationHeader = "Bee-Session-Key"
+export var authenticationHeader = "bee-session-key"
```
Comment (smatt989) [inline src/main/webapp/front-end/src/utilities.js:7]: ? why lowercase?

File: src/main/webapp/front-end/src/components/Register.jsx (hunk 2 above)
Comment (smatt989) [reply src/main/webapp/front-end/src/components/Register.jsx]: yea doesn't bother me.

File: src/main/webapp/front-end/src/reducer.js (hunk 3 above)
Comment (thomasweng15) [reply src/main/webapp/front-end/src/reducer.js:39]: I'm planning on nesting it, just kept it simple for now

File: src/main/webapp/front-end/src/reducer.js (hunk 4 above)
Comment (thomasweng15) [reply src/main/webapp/front-end/src/reducer.js]: fixed

File: src/main/webapp/front-end/src/utilities.js (hunk 5 above)
Comment (thomasweng15) [reply src/main/webapp/front-end/src/utilities.js:7]: it's undefined when uppercase even though it shows up as upper case in the Network Tab, I'll investigate further

File: src/main/webapp/front-end/src/components/Login.jsx (hunk 1 above)
Comment (thomasweng15) [reply src/main/webapp/front-end/src/components/Login.jsx]: fixed

File: src/main/webapp/front-end/src/utilities.js (hunk 5 above)
Comment (smatt989) [reply src/main/webapp/front-end/src/utilities.js:7]: yikes.  thats a bummer.  ok keep it i guess...

**Pull request merged**